iym/tasks.py

```diff
@@ -0,0 +1,349 @@
+from django.contrib import admin
+from iym import models
+from io import StringIO
+import petl as etl
+from decimal import Decimal
+from urllib.parse import urlencode
+from slugify import slugify
+from zipfile import ZipFile
+from django.conf import settings
+from django_q.tasks import async_task
+
+import os
+import csv
+import tempfile
+import requests
+import hashlib
+import base64
+import json
+import time
+import re
+import iym
+import datetime
+
+RE_END_YEAR = re.compile(r"/\d+")
+
+MEASURES = [
+    "Budget",
+    "AdjustmentBudget",
+    "April",
+    "May",
+    "June",
+    "July",
+    "August",
+    "September",
+    "October",
+    "November",
+    "December",
+    "January",
+    "February",
+    "March",
+    "Q1",
+    "Q2",
+    "Q3",
+    "Q4",
+]
+
+
+def authorise_upload(path, filename, userid, data_package_name, datastore_token):
+    md5 = hashlib.md5()
+    with open(path, "rb") as fh:
+        while True:
+            bytes = fh.read(1000)
+            if not bytes:
+                break
+            else:
+                md5.update(bytes)
+
+    md5_b64 = base64.b64encode(md5.digest())
+
+    authorize_upload_url = f"{settings.OPENSPENDING_HOST}/datastore/"
+    authorize_upload_payload = {
+        "metadata": {
+            "owner": userid,
+            "name": data_package_name,
+            "author": "Vulekamali",
+        },
+        "filedata": {
+            filename: {
+                "md5": md5_b64,
+                "name": filename,
+                "length": len(bytes),
```

Comment:
this is gonna be 1000 or 0 or somethiing in between, isn't it?

I think you want os.stat(file_name).st_size here

Reply:
done